subprojects/core/src/main/java/org/gradle/initialization/DefaultLegacyTypesSupport.java

```diff
@@ -63,14 +91,252 @@ private Set<String> readClassNames(String resourceName) {
         return classNames;
     }
 
+    public byte[] generateConventionImplementation(String className) {
```

Comment:
Looking at the formatting, we should do some cleanup. I'd prefer the code to look consistent with the existing asm code, like `AsmBackedClassGenerator` or `AdhocInterceptors`.
Regarding the content itself, I'm not proficient with asm to read and interpret the generated bytecode. If it does what we expect in test coverage (and does not blow up with a invalid class file format exception), then it's already a good indication about the validity.